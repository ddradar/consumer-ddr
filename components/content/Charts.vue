<template>
  <div class="tags has-addons" style="display: inline-block; margin: 0px">
    <span
      v-for="(level, i) in splitedLevels"
      :key="i"
      class="tag"
      :class="software.difficulties[splitedDificulties[i]].class"
      :title="software.difficulties[splitedDificulties[i]].name"
    >
      {{ level }}
    </span>
  </div>
</template>

<script lang="ts" setup>
import useSoftwareData from '~~/composables/useSoftwareData'

interface Props {
  slug: string
  levels: string
  difficulties: string
}

const _prop = defineProps<Props>()
const splitedLevels = computed(() => _prop.levels.split(','))
const splitedDificulties = computed(() =>
  _prop.difficulties.split(',').map((s) => parseInt(s, 10))
)

const { software } = await useSoftwareData(_prop.slug)
</script>
