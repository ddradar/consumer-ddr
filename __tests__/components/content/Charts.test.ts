import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import Charts from '~~/components/content/Charts.vue'

describe('components/content/Charts.vue', () => {
  const difficulties = [
    { name: 'BEGINNER', type: 'info' },
    { name: 'BASIC', type: 'warning' },
    { name: 'DIFFICULT', type: 'danger' },
    { name: 'EXPERT', type: 'success' },
    { name: 'CHALLENGE', type: 'primary' },
  ]
  test('renders badges', () => {
    const wrapper = mount(Charts, {
      global: { stubs: { Badge: true } },
      props: { difficulties, charts: '0,1,2,3,4', levels: '1,2,3,4,5' },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
