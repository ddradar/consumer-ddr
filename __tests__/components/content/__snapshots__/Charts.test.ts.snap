// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/content/Charts.vue > renders badges 1`] = `
<div
  data-v-app=""
>
  
  <badge-stub
    title="BEGINNER"
    type="info"
  />
  <badge-stub
    title="BASIC"
    type="warning"
  />
  <badge-stub
    title="DIFFICULT"
    type="danger"
  />
  <badge-stub
    title="EXPERT"
    type="success"
  />
  <badge-stub
    title="CHALLENGE"
    type="primary"
  />
  
</div>
`;
